File: app/src/at-nhatnguyen/java/asiantech/internship/summer/javacore/Bai17.java
package asiantech.internship.summer.javacore;
import java.util.Scanner;
public class Bai17 {
    public static int input(){
        Scanner scan= new Scanner(System.in);
        boolean check= false;
        int n=0;
        while(!check){
            System.out.print(" ");
            try{
                n= scan.nextInt();
                check= true;
            }catch(Exception e){
                System.out.println("Ban phai nhap so ");
                scan.nextLine();
            }
        }
        return (n);
    }
    private static float nhapFloat(){
        Scanner input= new Scanner(System.in);
        boolean check= false;
        float n=0;
        while(!check){
            System.out.print(" ");
            try{
                n= input.nextInt();
                check= true;
            }catch(Exception e){
                System.out.println("Ban phai nhap so! hay nhap lai...");
                input.nextLine();
            }
        }
        return (n);
    }
    private static int viTriMinFloat(float[] a, int n){
        float min= a[0];
        int key= 0;
        for(int j=0 ; j<n ; j++){
            if(min>a[j]){
                min= a[j];
                key= j;
            }
        }
        return (key);
    }
    private static float maxFloat(float[] a, int n){
        float max= a[0];
        for(int j=0 ; j<n ; j++){
            if(max<a[j]) max= a[j];
        }
        return (max);
    }
    public static void main(String[] args) {
        int n,i;
        System.out.println("Nhap n= ");
        n= input();
        float[] array= new float[n];
        for(i=0 ; i<n ; i++){
            System.out.println("Nhap phan tu thu " +(i+1)+" ");
            array[i]= nhapFloat();
        }
        i =0;
        System.out.println("Sap xep theo thu tu tang dan");
        while(i<n){
            System.out.println(" "+array[viTriMinFloat(array, n)]);
            array[viTriMinFloat(array, n)]= maxFloat(array, n);
            i++;
        }
    }
}

File: app/src/at-trinhnguyen/java/asiantech/internship/winter/javacore/congtrinh/Bai17.kt
package com.congtrinh
import com.congtrinh.a.nhap
import java.util.*


/*
Bài 17. Nhập số n và dãy các số thực a0 , a1 ,..., an-1.
Không đổi chỗ các phần tử và không dùng thêm mảng số thực nào khác
(có thể dùng mảng số nguyên nếu cần) hãy cho hiện trên màn hình dãy trên theo thứ tự tăng dần.
 */
object Bai17 {
    @JvmStatic
    fun main(args: Array<String>) {
        val scanner = Scanner(System.`in`)
        print("Enter n: ")
        val n = scanner.nextInt()
        val array = FloatArray(n)

        for (i in 0 until n) {
            println("Nhap phan tu thu ${i + 1}: ")
            array[i] = scanner.nextFloat()
        }
        println("Sap xep theo thu tu tang dan")
        for( j in 0 until n){
            println("${array[indexOfMinValue(array,n)]}")
            array[indexOfMinValue(array,n)]= maxValue(array,n)
        }
    }

    private fun indexOfMinValue(a: FloatArray, n: Int): Int {
        var min = a[0]
        var i = 0
        for (j in 0 until n) {
            if (min > a[j]) {
                min = a[j]
                i = j
            }
        }
        return i
    }

    private fun maxValue(a: FloatArray, n: Int): Float {
        var max = a[0]
        for (j in 0 until n) {
            if (max < a[j]) max = a[j]
        }
        return max
    }
}